### DISCUSSION TITLE
Using GPU for AutoMLForecast

### DISCUSSION THREAD

**Discussion by Avidar123** [Nixtla/mlforecast#403] [Q&A]
I'm training two models (AutoXGBoost and AutoLightGBM) using AutoMLForecast and want to use GPU. I am aware both models use Optuna for hyperparameter tuning. If I pass the following parameters:

For AutoXGBoost:
'tree_method': 'gpu_hist'
'predictor': 'gpu_predictor'
... rest of the params

For AutoLightGBM:
'device': 'gpu'
... rest of the params

Will this speed up the process and utilize the GPU, or could it cause issues in the backend, or would it make no difference ?

**Comment by jmoralez** [comment]
If the bottleneck is the training the it should help. If you're not training a lot of iterations then the bottleneck may be the data processing.
